x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java

```diff
@@ -1363,8 +1363,116 @@ public void testMetrics() {
         );
     }
 
+    public void testLookupJoin() {
+        assertFieldNames(
+            "FROM employees | KEEP languages | RENAME languages AS language_code | LOOKUP JOIN languages_lookup ON language_code",
+            Set.of("languages", "language_code", "language_code.*", "languages.*"),
```

Comment:
Please, group these based on the root field to keep the style of the test class: `"languages", "languages.*", "language_code", "language_code.*"`

Reply:
Done